torch/_tensor.py

```diff
@@ -455,12 +455,19 @@ def __setstate__(self, state):
         self.requires_grad, _, self._backward_hooks = state
 
     def __repr__(self, *, tensor_contents=None):
-        if has_torch_function_unary(self):
-            return handle_torch_function(
-                Tensor.__repr__, (self,), self, tensor_contents=tensor_contents
-            )
-        # All strings are unicode in Python 3.
-        return torch._tensor_str._str(self, tensor_contents=tensor_contents)
+        _in_recursive_repr = getattr(self, "_in_recursive_repr", 0)
+        if _in_recursive_repr > 0:
+            return f"<recursive tensor {id(self)}>"
+        self._in_recursive_repr = _in_recursive_repr + 1
+        try:
+            if has_torch_function_unary(self):
+                return handle_torch_function(
+                    Tensor.__repr__, (self,), self, tensor_contents=tensor_contents
+                )
+            # All strings are unicode in Python 3.
+            return torch._tensor_str._str(self, tensor_contents=tensor_contents)
+        finally:
+            self._in_recursive_repr = _in_recursive_repr
```

Comment:
not sure i follow why you need the try finally? if any of this throws an exception, then the repr results wont be important anyway?